packages/graphql-language-service-server/src/__tests__/MessageProcessor-test.js

```diff
@@ -0,0 +1,156 @@
+import {beforeEach, describe, it} from 'mocha';
+
+import * as handlers from '../MessageProcessor';
+import MockWatchmanClient from '../__mocks__/MockWatchmanClient';
+
+describe('MessageProcessor', () => {
+  const queryDir = `${__dirname}/__queries__`;
+
+  const textDocumentTestString = `
+  {
+    hero(episode: NEWHOPE){
+    }
+  }
+  `;
+
+  const initialDocument = {
+    textDocument: {
+      text: textDocumentTestString,
+      uri: `${queryDir}/test.graphql`,
+      version: 0,
+    },
+  };
+
+  async function setup() {
+    const params = {rootPath: __dirname};
+    const watchmanClient = new MockWatchmanClient();
+    const {capabilities} = await handlers.handleInitializeRequest(
+      params,
+      null,
+      null,
+      watchmanClient,
+    );
+
+    expect(capabilities.definitionProvider).to.equal(true);
+    expect(capabilities.completionProvider.resolveProvider).to.equal(true);
+    expect(capabilities.textDocumentSync).to.equal(1);
+
+    const result = await handlers.handleDidOpenOrSaveNotification(
+      initialDocument,
+    );
+    expect(result.uri).to.equal(initialDocument.textDocument.uri);
+
+    // Invalid query, diagnostics will show an error
+    expect(result.diagnostics.length).not.to.equal(0);
+  }
+
+  describe('setup', () => {
+    it('initializes properly', setup);
+  });
+
+  describe('fileLifecycle', () => {
+    beforeEach(setup);
+
+    it('runs completion requests', async () => {
+      const empty = {
+        textDocument: {text: '', uri: `${queryDir}/test2.graphql`, version: 0},
+      };
+      const out = await handlers.handleDidOpenOrSaveNotification(empty);
+      expect(out.uri).to.equal(empty.textDocument.uri);
+
+      const test = {
+        position: new Position(0, 0),
+        textDocument: empty.textDocument,
+      };
+
+      const expected = ['query', 'mutation', 'subscription', 'fragment', '{'];
+      const result = await handlers.handleCompletionRequest(test);
+      expect(result.items.length).to.equal(5);
+
+      for (const index in result.items) {
+        expect(expected).to.include(result.items[index].label);
+      }
+    });
+
+    // Doesn't work with mock watchman client
+    it.skip('runs definition requests', async () => {
+      const validQuery = `
+      {
+        hero(episode: EMPIRE){
+          ...testFragment
+        }
+      }
+      `;
+
+      const newDocument = {
+        textDocument: {
+          text: validQuery,
+          uri: `${queryDir}/test3.graphql`,
+          version: 0,
+        },
+      };
+
+      await handlers.handleDidOpenOrSaveNotification(newDocument);
+
+      const test = {
+        position: new Position(3, 15),
+        textDocument: newDocument.textDocument,
+      };
+
+      const result = await handlers.handleDefinitionRequest(test);
+      expect(result[0].uri).to.equal(`file://${queryDir}/testFragment.graphql`);
+    });
+
+    it('updates cache when the file is changed', async () => {
```

Comment:
This doesn't really check the cache though, right? We should change the description to match what the test is actually testing - perhaps `properly changes the file cache with didChange notification` or something similar may fit better.